src/main/java/org/wahlzeit_revisited/model/PhotoStatus.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2006-2009 by Dirk Riehle, http://dirkriehle.com
+ *
+ * This file is part of the Wahlzeit photo rating application.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+package org.wahlzeit_revisited.model;
+
+import org.wahlzeit.utils.EnumValue;
+
+/**
+ * The PhotoStatus of a Photo captures its state in the system.
+ * A photo may be visible or invisible, it may have been flagged, and it may have been deleted.
+ * These states are not mutually exclusive, hence the bitset simulation in this class.
+ */
+public enum PhotoStatus implements EnumValue {
+
+    /**
+     *
+     */
+
+    VISIBLE(0), // no bit set
+    INVISIBLE(1), // only invisible bit set
+    FLAGGED(2), // flagged without or with invisible bit set
+    MODERATED(3),
+    DELETED(4);
+
+    /**
+     *
+     */
+    private static final int INVISIBLE_BIT = 0;
+    private static final int FLAGGED_BIT = 1;
+    private static final int MODERATED_BIT = 2;
+    private static final int DELETED_BIT = 3;
+
+    /**
+     * All possible states of PhotoStatus
+     */
+    private static PhotoStatus[] allValues = {
+            VISIBLE, INVISIBLE, FLAGGED, MODERATED, DELETED,
+    };
+
+    /**
+     *
+     */
+    public static PhotoStatus getFromInt(int myValue) throws IllegalArgumentException {
+        if ((myValue >= 0) && (myValue <= 15)) {
+            return allValues[myValue];
+        }
+
+        throw new IllegalArgumentException("invalid PhotoStatus int: " + myValue);
+    }
+
+    /**
+     *
+     */
+    private static String[] valueNames = {
```

Comment:
Here are also int mixed with string, not sure if we should improve that by using a string-based enum

Reply:
I am not quite sure about this class at all. 

It uses it's int value as an bitmap and handles it's state with the according bit operations.

Reply:
There are basically 4 bits (almost) all independent of each other. So mapping to a single string doesn't work, a bit set would work and in my original implementation I am emulating it. Are bit-set based enums? It could be a regular value type, doesn't have to be enum. I'm not sure but I'd prefer to keep it as a value object (but probably could be a regular object).